internal/domains/archiver.go

```diff
@@ -39,20 +37,19 @@ func (d *ArchiverDomain) DownloadBookmarkArchive(book model.Bookmark) (*model.Bo
 	return &result, nil
 }
 
-func (d *ArchiverDomain) GetBookmarkArchive(book model.Bookmark) error {
-	archivePath := filepath.Join(d.dataDir, "archive", strconv.Itoa(book.ID))
+func (d *ArchiverDomain) GetBookmarkArchive(book *model.BookmarkDTO) (*warc.Archive, error) {
+	archivePath := model.GetArchivePath(book)
 
-	info, err := os.Stat(archivePath)
-	if !os.IsNotExist(err) && !info.IsDir() {
-		return fmt.Errorf("archive not found")
+	if !d.deps.Domains.Storage.FileExists(archivePath) {
+		return nil, fmt.Errorf("archive for bookmark %d doesn't exist", book.ID)
 	}
 
-	return nil
+	// FIXME: This only works in local filesystem
```

Comment:
Right now it can work as it is, since we don't allow using non-local filesystems, and this will require a bigger refactor (and I don't want to make this PR larger)

Reply:
https://github.com/go-shiori/shiori/issues/805 👍